src/types/dict.rs

```diff
@@ -517,6 +517,16 @@ impl<'py> PyDictMethods<'py> for Bound<'py, PyDict> {
     }
 }
 
+impl<'py> Bound<'py, PyDict> {
+    /// Iterates over the contents of this dictionary without incrementing reference counts.
+    ///
+    /// # Safety
+    /// It must be known that this dictionary will not be modified during iteration.
+    pub(crate) unsafe fn iter_borrowed<'a>(&'a self) -> BorrowedDictIter<'a, 'py> {
+        BorrowedDictIter::new(self)
+    }
+}
```

Comment:
The purpose of adding this crate-private API was that I wanted to keep keyword argument processing using `Borrowed` for both the "fastcall" and default calling conventions. As far as I am aware, argument processing can safely process a `kwargs` dict in this way because it's the only referee of the kwargs dict.